test/util/BenchmarkMeasurementContainerHelpers.h

```diff
@@ -22,6 +24,22 @@ CPP_template(typename Type)(requires ad_utility::SimilarToAnyTypeIn<
                             Type, ad_benchmark::ResultTable::EntryType>) Type
     createDummyValueEntryType();
 
+namespace detail {
+// Helper for the `doForTypeInResultTableEntryType` function (see below).
+template <typename Function>
+struct FunctionCaller {
+  Function& function;
+
```

Comment:
This automatically gets simpler (again a small lambda)
if the `forEachTypeInTemplateType` gets refactored to use `type_identity`..

Reply:
Addressed